userspace/libsinsp/threadinfo.h

```diff
@@ -1024,4 +1027,6 @@ class SINSP_PUBLIC sinsp_thread_manager: public libsinsp::state::table<int64_t>
 	int32_t m_n_main_thread_lookups = 0;
 	int32_t m_max_n_proc_lookups = -1;
 	int32_t m_max_n_proc_socket_lookups = -1;
+
+	std::shared_ptr<libsinsp::state::dynamic_struct::field_infos> m_fdtable_dyn_fields;
```

Comment:
Note -- having a single shared definition list for dynamic fields maintained by a thread info table gives us the guarantee that all file descriptor sub-tables, across all thread infos of the same thread manager, will have the same dynamic fields definitions.

From the perspective of plugins, this means that once a dynamic field accessor is obtained at init time, it can be used freely across all the file descriptor sub-tables it will have access to later on (this is covered by a test too). This enables safe dynamic enrichment of file descriptor infos.